src/test/java/bio/terra/app/controller/SnapshotOperationTest.java

```diff
@@ -321,40 +316,29 @@ private BigQueryProject bigQueryProjectForDatasetName(String datasetName) {
         return BigQueryProject.get(dataProject.getGoogleProjectId());
     }
 
-    private void loadData(String datasetName,
+    private void loadData(String datasetId,
                           String tableName,
                           String resourcePath,
-                          FormatOptions options) throws Exception {
-        String snapshotName = PDAO_PREFIX + datasetName;
-        String location = "US";
-        TableId tableId = TableId.of(snapshotName, tableName);
-        BigQueryProject bigQueryProject = bigQueryProjectForDatasetName(datasetName);
-        BigQuery bigQuery = bigQueryProject.getBigQuery();
+                          IngestRequestModel.FormatEnum format) throws Exception {
 
-        WriteChannelConfiguration writeChannelConfiguration =
-            WriteChannelConfiguration.newBuilder(tableId).setFormatOptions(options).build();
+        String bucket = testConfig.getIngestbucket();
```

Comment:
Perhaps add a comment describing why this section being refactored was necessary.

Reply:
Sure! This test is poking at a dataset purely through the API, so it never gets a handle on a `Dataset` instance. The names of the raw / SD tables aren't exposed over the API (should they be?). That causes problems when the test tried to load data into the table, because the only name it can find is the name of the live view, and it's not possible to insert into a view.

The new logic uses the ingest API to perform the data load and avoid the problem. The same pattern is used in the `BigQueryPdaoTest`.

Reply:
Thanks, good point. I don't think we want to expose those and should consider them "internal" to the repo so that we can alter how we use them without breaking things for people consuming them.